realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1649,6 +1677,129 @@ public void distinctAsync_notIndexedLinkedFields() {
         }
     }
 
+    @Test
+    public void distinctMultiArgs() {
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10; // must be greater than 1
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        //Simon Ask Ulsnes "I would expect it to be 25, because I would expect it to look for unique ​_combinations_​ or tuples."
+        RealmResults<AnnotationIndexTypes> distinctMulti = realm.where(AnnotationIndexTypes.class).findAll().distinct(AnnotationIndexTypes.INDEX_FIELDS);
+        assertEquals(numberOfBlocks, distinctMulti.size());
+    }
+
+    @Test
+    public void distinctMultiArgs_switchedFieldsOrder() {
+        final long numberOfBlocks = 25; // must be greater than 1
+        populateForFieldsOrderSwitch(realm, numberOfBlocks);
+
+        RealmResults<AnnotationIndexTypes> distinctStringLong = realm.where(AnnotationIndexTypes.class).distinct(AnnotationIndexTypes.FIELDS_SWITCH_ORDER_STRING_LONG);
+        RealmResults<AnnotationIndexTypes> distinctLongString = realm.where(AnnotationIndexTypes.class).distinct(AnnotationIndexTypes.FIELDS_SWITCH_ORDER_LONG_STRING);
+        assertEquals(4, distinctStringLong.size());
+        assertEquals(4, distinctLongString.size());
+        assertEquals(distinctStringLong.size(), distinctLongString.size());
+    }
+
+    @Test
+    public void distinctMultiArgs_singleField() {
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10;
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        RealmResults<AnnotationIndexTypes> distinctMulti = realm.where(AnnotationIndexTypes.class).findAll().distinct(new String[]{AnnotationIndexTypes.FIELD_INDEX_BOOL});
```

Comment:
I would refactor all these tests so to:

``` 
RealmResults<AnnotationIndexTypes> results = realm.where(AnnotationIndexTypes.class).findAll();
RealmResults<AnnotationIndexTypes> distinctMulti = results.distinct();
```

That way it is a lot easier to see that it is RealmResults being tested.